cpp/src/parquet/arrow/reader_writer_benchmark.cc

```diff
@@ -197,6 +197,52 @@ BENCHMARK_TEMPLATE2(BM_WriteColumn, true, DoubleType);
 BENCHMARK_TEMPLATE2(BM_WriteColumn, false, BooleanType);
 BENCHMARK_TEMPLATE2(BM_WriteColumn, true, BooleanType);
 
+int32_t kInfiniteUniqueValues = -1;
+
+std::shared_ptr<::arrow::Table> RandomStringTable(int64_t length, int64_t unique_values,
+                                                  int64_t null_percentage) {
+  std::shared_ptr<::arrow::DataType> type = ::arrow::utf8();
+  std::shared_ptr<::arrow::Array> arr;
+  ::arrow::random::RandomArrayGenerator generator(500);
+  double null_probability = static_cast<double>(null_percentage) / 100.0;
+  if (unique_values == kInfiniteUniqueValues) {
+    arr = generator.String(length, /*min_length=*/3, /*max_length=*/32,
+                           /*null_probability=*/null_probability);
+  } else {
+    arr = generator.StringWithRepeats(length, /*unique=*/unique_values,
+                                      /*min_length=*/3, /*max_length=*/32,
+                                      /*null_probability=*/null_probability);
+  }
+  return ::arrow::Table::Make(
+      ::arrow::schema({::arrow::field("column", type, null_percentage > 0)}), {arr});
+}
+
+static void BM_WriteBinaryColumn(::benchmark::State& state) {
+  std::shared_ptr<::arrow::Table> table =
+      RandomStringTable(BENCHMARK_SIZE, state.range(1), state.range(0));
+
+  while (state.KeepRunning()) {
+    auto output = CreateOutputStream();
+    EXIT_NOT_OK(
+        WriteTable(*table, ::arrow::default_memory_pool(), output, BENCHMARK_SIZE));
+  }
+
+  int64_t total_bytes = table->column(0)->chunk(0)->data()->buffers[1]->size();
+  state.SetItemsProcessed(BENCHMARK_SIZE * state.iterations());
+  state.SetBytesProcessed(total_bytes * state.iterations());
+}
+
+BENCHMARK(BM_WriteBinaryColumn)
+    ->Args({/*null_probability*/ 0, /*unique_values*/ 2})
+    ->Args({/*null_probability*/ 0, /*unique_values*/ 32})
+    ->Args({/*null_probability*/ 0, /*unique_values*/ kInfiniteUniqueValues})
+    ->Args({/*null_probability*/ 1, /*unique_values*/ 32})
+    ->Args({/*null_probability*/ 50, /*unique_values*/ 32})
+    ->Args({/*null_probability*/ 99, /*unique_values*/ 32})
+    ->Args({/*null_probability*/ 1, /*unique_values*/ kInfiniteUniqueValues})
+    ->Args({/*null_probability*/ 50, /*unique_values*/ kInfiniteUniqueValues})
+    ->Args({/*null_probability*/ 99, /*unique_values*/ kInfiniteUniqueValues});
```

Comment:
Can you use `ArgNames` to make the arguments more descriptive in the output?

Reply:
Good idea. If I do the same to the other benchmarks in the file, would that break the continuity of their results in conbench?

Reply:
I'm not sure, though I think we can live with that in any case.